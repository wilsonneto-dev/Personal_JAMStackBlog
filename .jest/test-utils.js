import { render } from '@testing-library/react';

import { ThemeProvider } from 'styled-components';
import defaultTheme from 'styles/themes/default';

const WithTheme = ({ children }) => {
  return <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>;
};

export const renderWithTheme = (ui, options) =>
  render(ui, { wrapper: WithTheme, ...options });
