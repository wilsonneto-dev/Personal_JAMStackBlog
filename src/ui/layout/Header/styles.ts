import styled from 'styled-components';

export const Wrapper = styled.header`
  height: 8rem;
  background-color: ${({ theme }) => theme.colors.featuredBackground};
`;

export const InnerWrapper = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  width: min(${(props) => props.theme.sizes.box}, 100%);
  padding: 0 1rem;
  margin: 0 auto;
  height: 100%;
`;

export const Logo = styled.img`
  height: 4rem;
`;

export const SocialIcon = styled.a`
  display: inline-block;
  img {
    height: 1.6rem;
    margin-left: 0.5rem;
  }
`;
