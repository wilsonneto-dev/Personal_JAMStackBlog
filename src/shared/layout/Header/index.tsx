import Link from 'next/link';
import * as S from './styles';

const Header = () => (
  <S.Wrapper>
    <S.InnerWrapper>
      <div>
        <Link href="/">
          <a>
            <S.Logo
              src="/img/wilsonneto.png"
              alt="Logo Wilson Neto"
              role="logo"
            />
          </a>
        </Link>
      </div>
      <div>
        <S.SocialIcon target="_blank" href="/" role="social-link">
          <img src="/img/ic-github.svg" alt="Wilson Neto - Github" />
        </S.SocialIcon>
        <S.SocialIcon target="_blank" href="/" role="social-link">
          <img src="/img/ic-linkedin.svg" alt="Wilson Neto - Linkedin" />
        </S.SocialIcon>
      </div>
    </S.InnerWrapper>
  </S.Wrapper>
);

export default Header;
