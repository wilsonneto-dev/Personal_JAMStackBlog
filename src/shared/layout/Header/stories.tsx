import { Story, Meta } from '@storybook/react/types-6-0';
import Header from '.';

export default {
  title: 'Layout/Header',
  component: Header
} as Meta;

export const Default: Story = () => <Header />;
