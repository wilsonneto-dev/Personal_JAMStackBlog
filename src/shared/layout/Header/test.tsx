import { screen } from '@testing-library/react';
import { renderWithTheme } from '../../../../.jest/test-utils';
import Header from '.';
// import { getByRole } from '@testing-library/react';

describe('<Header />', () => {
  it('should render the logo', () => {
    renderWithTheme(<Header />);
    expect(screen.getByRole('logo')).toBeInTheDocument();
  });

  it('should render social links', () => {
    renderWithTheme(<Header />);
    expect(screen.getAllByRole('social-link').length).toBeGreaterThan(0);
  });
});
