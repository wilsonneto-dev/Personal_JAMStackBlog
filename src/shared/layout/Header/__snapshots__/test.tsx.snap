// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render the heading 1`] = `
.c0 {
  height: 8rem;
  background-color: #612F74;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: min(100em,100%);
  padding: 0 1rem;
  margin: 0 auto;
  height: 100%;
}

.c2 {
  height: 4rem;
}

<header
  class="c0"
>
  <div
    class="c1"
  >
    <div>
      <a
        href="/"
      >
        <img
          alt="Logo Wilson Neto"
          class="c2"
          role="logo"
          src="/img/wilsonneto.png"
        />
      </a>
    </div>
    <div />
  </div>
</header>
`;
